openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+from collections import Iterable
+from numbers import Real, Integral
+import copy
+import sys
+
+import numpy as np
+
+import openmc.checkvalue as cv
+
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class EnergyGroups(object):
+    """An energy groups structure used for multi-group cross-sections.
+
+    Parameters
+    ----------
+    group_edges : ndarray
+        The energy group boundaries [MeV]
+    num_groups : Integral
```

Comment:
Agreed. The `_num_groups` is a relic from an early implementation and definitely is no longer needed, so it has been removed.